src/server/common_services/auth_server.js

```diff
@@ -670,14 +673,16 @@ function _get_auth_info(account, system, authorized_by, role, extra) {
  * @returns  {Promise<boolean>} true if the account has permission to perform the action on the bucket
  */
 async function has_bucket_action_permission(bucket, account, action, req_query, bucket_path = "") {
-    dbg.log1('has_bucket_action_permission:', bucket.name, account.email, bucket.owner_account.email);
+    dbg.info('has_bucket_action_permission:', bucket.name.unwrap(),bucket.system.owner.email.unwrap(), account.email.unwrap(), bucket.owner_account.email.unwrap());
```

Comment:
Used Command
```
sh-4.2$ aws s3 ls --endpoint-url http://s3-endpoint-proxy.openshift-storage-client.svc.cluster.local
2025-03-21 09:07:05 client-bucket // client owns this bucket
2025-03-19 12:05:05 first.bucket // admin owns
2025-03-21 08:56:26 test-bucket // admin owns
```
```[INFO] core.server.common_services.auth_server:: has_bucket_action_permission: test-bucket admin@noobaa.io storageconsumer-7a3ae305-b4cf-4547-a5af-aa1f91dbe79a admin@noobaa.io```

According to the logic here, this bucket should not have been listed in the output
since account.email (`storageconsumer-7a3ae305-b4cf-4547-a5af-aa1f91dbe79a `) does not equal bucket.owner_account.email (`admin@noobaa.io`)
@nimrod-becker @jackyalbo Any thoughts?